pdf/renderer.go

```diff
@@ -940,6 +950,67 @@ func (w *pdfPageWriter) DrawImage(img image.Image, enc canvas.ImageEncoding, m c
 }
 
 func (w *pdfPageWriter) embedImage(img image.Image, enc canvas.ImageEncoding) pdfName {
+	var stream pdfStream
+	if i, ok := img.(canvas.Image); ok && i.Mimetype == canvas.ImageJPEG && len(i.Bytes) > 0 {
+		stream = w.jpegStream(i)
+	} else {
+		stream = w.imageStream(img)
+	}
+
+	ref := w.pdf.writeObject(stream)
+	if _, ok := w.resources["XObject"]; !ok {
+		w.resources["XObject"] = pdfDict{}
+	}
+	name := pdfName(fmt.Sprintf("Im%d", len(w.resources["XObject"].(pdfDict))))
+	w.resources["XObject"].(pdfDict)[name] = ref
+	return name
+}
+
+func (w *pdfPageWriter) jpegStream(img canvas.Image) pdfStream {
+	// ignore progressive jpeg (contains 0xff 0xc2 marker)
+	markerStarted := false
+	for _, b := range img.Bytes {
+		if markerStarted && b == 0xc2 {
+			// fallback to generic imageStream
+			return w.imageStream(img)
+		}
+		markerStarted = (b == 0xff)
+	}
+
+	size := img.Bounds().Size()
+	dict := pdfDict{
```

Comment:
Where do we set the `SMask` key for the mask?

Reply:
As far as I know, there is no transparency support on jpeg files. So all the pixels will be opaque and `hasMask` will be false (if re-implemented like in `imageStream`)

Reply:
You are right! It's still in `imageStream`, where it should